Docs update (#653)
elastic/apm-agent-android: docs/reference/edot-android/automatic-instrumentation.md

1. Make sure the adapter is added.
2. You can find the dependencies needed in the [instrumentation's README file](https://github.com/open-telemetry/opentelemetry-android/tree/main/instrumentation/httpurlconnection#project-dependencies). The same will be the case for any other instrumentation.
3. The instrumentations that require a `byteBuddy` dependency, do bytecode weaving, as explained in [compilation behavior](#compilation-behavior). An extra plugin named `net.bytebuddy.byte-buddy-gradle-plugin` is required to make this work, as shown [here](https://github.com/open-telemetry/opentelemetry-android/tree/main/instrumentation/httpurlconnection#byte-buddy-compilation-plugin). However, EDOT Android installs this extra plugin on your behalf, so there's no need for you to do so manually.

### Compilation performance

> As explained in [compilation behavior](#compilation-behavior), the instrumentations that perform bytecode weaving can increase the

Reviewer: ```suggestion
As explained in [compilation behavior](#compilation-behavior), the instrumentations that perform bytecode weaving might increase the
```

> compilation times, which might not be needed all the time. Because of this, EDOT Android provides a [configuration param](#automatic-instrumentation-configuration)

Reviewer: > which might not be needed all the time.

Bytecode weaving is not needed all the time? If that's the case, I suggest this rewrite:

```suggestion
compilation times. Because bytecode weaving might not be necessary, EDOT Android provides a [configuration param](#automatic-instrumentation-configuration)
```

Author: > Bytecode weaving is not needed all the time?

Well, maybe the wording I used is not the right one now that I think about it. It's always necessary to make the instrumentation work properly, but it's just that adding observability to an app might not always be necessary (like during development for example), and rather is more of a tool for apps in production. And since developers tend to build an app multiple times while adding code changes to check how their new code behaves, it might be annoying to slow down the compilation in these cases because of a feature that they might not want to use during development, such as observability. Let me see how I can rephrase it a little bit.

Author: I've just removed the confusing part. I noticed that the message I wanted to convey is still there without that part, and it's more straightforward to read as well.

> that allows to select specific build types for which the bytecode instrumentation will be disabled. However, said configuration

Reviewer: ```suggestion
that allows to select specific build types for which the bytecode instrumentation will be turned off. This configuration
```

only works for EDOT Android's [supported instrumentations](#supported-instrumentations), so it won't have any effect on

> instrumentations added via the OTel Android instrumentations adapter.

Reviewer: ```suggestion
instrumentations added through the OTel Android instrumentations adapter.
```

In order to select the build variant to run the bytecode instrumentation for an OTel Android instrumentation, you must
add its `byteBuddy` dependency using the variant-specific byteBuddy name, such as `releaseByteBuddy` or `debugByteBuddy`,
depending on which variant you'd like to enable the bytecode instrumentation for.

> Following the example above, if we'd like to install the `HttpURLConnection` instrumentation only for our app's `release` build type,
> we can change the way we add its byteBuddy dependency as shown below:

Reviewer: ```suggestion
Following the previous example, if you want to install the `HttpURLConnection` instrumentation only for your app's `release` build type,
you can change the way we add its byteBuddy dependency as shown here:
```
